### DISCUSSION TITLE
Guided Exercise - React App - Emoji Fun

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#169] [Tasks]
Let's create a React app called "Emoji Fun" where users can select different emojis and see them displayed in a fun and creative way. This exercise will cover the concepts of state variables, props, and child components. Here are the detailed step-by-step instructions:

Step 1: Setup
- Open your favorite text editor and create a new directory for your project.
- Initialize a new React app using Create React App by running the command: `npx create-react-app emoji-fun`.
- Navigate into the project directory: `cd emoji-fun`.
- Start the development server: `npm start`.
- Open your browser and go to `http://localhost:3000` to see the default React app running.

Step 2: Clear the default project
- Open the `src/App.js` file and remove the existing code.
- Replace it with the following code:
```jsx
import React from 'react';

function App() {
  return (
    <div>
      <h1>Emoji Fun</h1>
    </div>
  );
}

export default App;
```

Step 3: Create EmojiSelector component
- Create a new file called `EmojiSelector.js` in the `src` directory.
- Open `EmojiSelector.js` and add the following code:
```jsx
import React from 'react';

function EmojiSelector({ onSelect }) {
  const emojis = ['😄', '🥳', '🎉', '🎈', '🤩', '😎'];

  return (
    <div>
      <h2>Select an Emoji</h2>
      <div className="emoji-container">
        {emojis.map(emoji => (
          <span
            key={emoji}
            className="emoji"
            onClick={() => onSelect(emoji)}
          >
            {emoji}
          </span>
        ))}
      </div>
    </div>
  );
}

export default EmojiSelector;
```

Step 4: Import and use EmojiSelector in App.js
- Open `src/App.js` and modify the code as follows:
```jsx
import React, { useState } from 'react';
import EmojiSelector from './EmojiSelector';

function App() {
  const [selectedEmoji, setSelectedEmoji] = useState(null);

  const handleEmojiSelect = (emoji) => {
    setSelectedEmoji(emoji);
  };

  return (
    <div>
      <h1>Emoji Fun</h1>
      {selectedEmoji && (
        <div>
          <h2>You selected:</h2>
          <span className="emoji">{selectedEmoji}</span>
        </div>
      )}
      <EmojiSelector onSelect={handleEmojiSelect} />
    </div>
  );
}

export default App;
```

Step 5: Add some styles
- Open the `src/App.css` file and replace the existing styles with the following code:
```css
body {
  font-family: Arial, sans-serif;
  text-align: center;
}

h1 {
  color: #333;
}

h2 {
  margin-top: 2rem;
  color: #666;
}

.emoji-container {
  display: flex;
  justify-content: center;
}

.emoji {
  font-size: 2rem;
  margin: 0 0.5rem;
  cursor: pointer;
}

.emoji:hover {
  transform: scale(1.2);
}
```

Step 6: Run the app
- Make sure the development server is running (`npm start`).
- Open your browser and go to `http://localhost:3000`.
- You should see the "Emoji Fun" heading and the emoji selector.
- Click on any emoji, and it should appear below the selector.

Congratulations! You have created the "Emoji Fun" React app. It allows users to select emojis and displays the selected emoji below the selector. Feel free to customize the styles, add more emojis, or add additional features to make it even more amusing and fun to work with. Happy coding!

**Comment by bipin-coded** [comment]
![Screenshot](https://github.com/akash-coded/mern/assets/126976134/e58352d7-050e-44c6-b442-f77f8ca8bbae)

**Reply by sherwinsamuel32** [reply]
![Screenshot (71)](https://github.com/akash-coded/mern/assets/129487106/a70a6ff3-4b2e-4aea-b39d-7050a745dee9)

**Comment by Ruchiarth** [comment]
![image](https://github.com/akash-coded/mern/assets/126975575/75edc2a6-faf2-4941-82f1-ae29b48a1df1)

**Comment by vk0027** [comment]
![image](https://github.com/akash-coded/mern/assets/123943655/2ca3fed6-5eea-4113-9f1c-301743a1ca51)

**Comment by Ravi-710** [comment]
![Screenshot (85)](https://github.com/akash-coded/mern/assets/98272814/a3bcd6ab-bd62-4640-95f8-6aefc850e310)

**Comment by Karthikkevin** [comment]
![Screenshot (149)](https://github.com/akash-coded/mern/assets/33948674/5378dc74-ca84-40c6-9b57-4cb1d39e2afc)

**Comment by SagarTChandrakanth** [comment]
![Screenshot from 2023-07-08 20-47-29](https://github.com/akash-coded/mern/assets/110445544/b4d81baa-0188-49b3-a08f-bded6878ff6a)

**Comment by jiten207** [comment]
Emoji Fun
Select an Emoji
😄🥳🎉🎈🤩😎

**Reply by jiten207** [reply]
Emoji Fun
You selected:
😎
Select an Emoji
😄🥳🎉🎈🤩😎

**Reply by jiten207** [reply]
![image](https://github.com/akash-coded/mern/assets/27985322/e48b55ab-9568-4fc8-bd21-b8676ecae355)

**Comment by mightyrsm** [comment]
<img width="410" alt="Screenshot 2023-07-08 at 8 50 35 PM" src="https://github.com/akash-coded/mern/assets/136921146/16c78460-3581-47c4-a7ae-df6474a31a06">

**Comment by sarwanuser** [comment]
![image](https://github.com/akash-coded/mern/assets/25919242/7dd6349b-56a3-42d8-a453-35fe8e350b85)

**Comment by madhuriS3008** [comment]
![image](https://github.com/akash-coded/mern/assets/85974647/ec46a437-97e1-48df-8eb7-253520219f31)

**Comment by pooja744** [comment]
![image](https://github.com/akash-coded/mern/assets/138996416/1c1c6cfe-c965-4ec5-ab07-66d4662bd5aa)

**Comment by Sagar1006-dev** [comment]
![2023-07-08](https://github.com/akash-coded/mern/assets/129708548/e0a00ff6-3348-4f05-bfaa-f745a0fdc5fa)

**Comment by chukku10** [comment]
![emoji fun](https://github.com/akash-coded/mern/assets/127116126/5545c579-1f29-4d3a-aae3-8f4b168690c6)

**Comment by nandini-naik09** [comment]
![image](https://github.com/akash-coded/mern/assets/133530727/441d83a1-6233-49b9-a254-5294b960a4fc)

**Comment by kkpai** [comment]
* + C 1 © http://localhost:3000
Emoji Fun
Select an Emoii

**Comment by pawarprathmesh333** [comment]
![image](https://github.com/akash-coded/mern/assets/71115267/aed55d18-1de8-45de-8020-18a23600d9e9)

**Comment by bhargavg2009** [comment]
![reac1](https://github.com/akash-coded/mern/assets/7472991/522731e8-f044-45b2-a53b-4ce93c1cae65)

**Comment by Sheetal-dh** [comment]
![image](https://github.com/akash-coded/mern/assets/91683804/9e3f20d1-0718-41b8-a608-b0adc7cef815)

**Comment by biswa348** [comment]
![react task](https://github.com/akash-coded/mern/assets/93532317/b5d74ec9-0526-4ed7-b2ac-7ea2d64a5411)

**Comment by anoop-manukumar** [comment]
![emoji](https://github.com/akash-coded/mern/assets/138996605/775fb515-92c0-4c74-97b9-4af1682d362f)

**Comment by KailashGareja** [comment]
![image](https://github.com/akash-coded/mern/assets/136106221/8fea3866-46ac-4370-980e-ccc95a74fc9a)

**Comment by Ananthu165** [comment]
![image](https://github.com/akash-coded/mern/assets/138997681/4eb0adec-e6d8-4917-baa5-d126bb02e39c)

**Comment by basu-cse** [comment]
![react1](https://github.com/akash-coded/mern/assets/95092878/bf6afb24-14e3-415b-a523-ca5cc3eb020d)

**Reply by parameshkomati** [reply]
![image](https://github.com/akash-coded/mern/assets/47549898/c60b0592-1e96-4c17-8e1c-c5e1c789ff8e)

**Comment by sahidraza** [comment]
![image](https://github.com/akash-coded/mern/assets/114484239/d3a17794-c9b3-4286-82fb-27770443c8ce)

**Comment by anantsinha7798** [comment]
![image](https://github.com/akash-coded/mern/assets/52276093/e466bcb7-ac42-470b-ac89-89401328e6a4)

**Comment by priyamallireddy** [comment]
<img width="960" alt="Screenshot 2023-07-08 214624" src="https://github.com/akash-coded/mern/assets/137372978/67a4e498-219f-4ee2-a168-ecedd40eeffa">

**Comment by nanditha-prabhakar** [comment]
![Screenshot (4)](https://github.com/akash-coded/mern/assets/126976487/5f755d53-66ce-434e-a6dd-84c98933ca44)

**Comment by Rutud2120** [comment]
![image](https://github.com/akash-coded/mern/assets/104456697/0f1ace46-1db8-4e6c-93ff-126cf2db9bee)

**Comment by mownica-p** [comment]
![image](https://github.com/akash-coded/mern/assets/126975660/f9e4ee4b-f95a-45c2-8051-a2f169a93a25)

**Comment by PragavathiVadivel** [comment]
![Emoji Fun](https://github.com/akash-coded/mern/assets/137114388/6ac6c1ce-a944-4b80-a191-c705223ca008)

**Comment by KavyaKetha** [comment]
![image](https://github.com/akash-coded/mern/assets/12147689/ea0dc519-6d9a-4e41-b183-09b743cdae1e)

**Comment by Sanket866** [comment]
![2](https://github.com/akash-coded/mern/assets/120538359/06ac741d-2794-43a0-95b1-415d4bfb4977)

**Comment by tinapatil13** [comment]
![discussion169](https://github.com/akash-coded/mern/assets/88702583/3a41b665-2b3f-432a-94a9-e233552e4f57)

**Comment by starvikas1988** [comment]
![image](https://github.com/akash-coded/mern/assets/70792693/139727f9-40a5-4b17-bbd3-3602a6e5d3ba)

**Comment by RevathiSundaramoorthy** [comment]
![image](https://github.com/akash-coded/mern/assets/126975777/82b6dde4-dfd5-4a6a-b5b1-be0043fc7a4c)

**Comment by sam2elect** [comment]
![image](https://github.com/akash-coded/mern/assets/85131631/3c34c937-5116-461a-a5c6-90d63d6fce06)
